opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -138,15 +138,119 @@ def __repr__(self):
         asyncio.run(main())
 """
 
+import typing
+
+from contextlib import contextmanager
+
 from .base_context import BaseRuntimeContext
 
 __all__ = ["Context"]
 
 try:
     from .async_context import AsyncRuntimeContext
 
-    Context = AsyncRuntimeContext()  # type: BaseRuntimeContext
+    _CONTEXT = AsyncRuntimeContext()  # type: BaseRuntimeContext
 except ImportError:
     from .thread_local_context import ThreadLocalRuntimeContext
 
-    Context = ThreadLocalRuntimeContext()
+    _CONTEXT = ThreadLocalRuntimeContext()
+
+
+class Context:
+    def __init__(self) -> None:
+        self.snapshot = _CONTEXT.snapshot()
```

Comment:
As I understood it, the point of using threadlocals and `ContextVars` was to avoid this kind of manual context management. If the context is backed by a threadlocal we should get a new copy for each thread we spawn, if it's backed by a `ContextVar` we should get a new copy for each new async context. When would we need to copy this ourselves?

Reply:
Correct, a new copy would be made available within a new thread or async context. But as @mauriciovasquezbernal's reply says, the spec suggests in the scope example: https://github.com/open-telemetry/oteps/blob/master/text/0066-separate-context-propagation.md#the-scope-of-current-context get_current returns an immutable Context.